### DISCUSSION TITLE
Plural acronyms

### DISCUSSION THREAD

**Discussion by dylanirion** [rchaput/acronyms#30] [Q&A]
Thank you for a great addition to the quarto ecosystem!
I think a useful addition might be to allow "pluralizing" acronyms, unless it's already possible and I've missed it?

Is there a more elegant solution than something like below?

```{lua}
styles["long-short-plural"] = function(acronym, insert_links, is_first_use)
  local text
  if is_first_use then
      text = acronym.longname .. (string.sub(acronym.longname, -1) == "s" and "es" or "s" ) .. " (" .. acronym.shortname .. "s" .. ")"
  else
      text = acronym.shortname .. (string.sub(acronym.longname, -1) == "s" and "es" or "s" )
  end

  return create_element(text, acronym.key, insert_links)
end
```

Happy to submit a PR if you think this is worthwhile!

**Comment by rchaput** [comment]
Hi,

Thanks for the appreciation!

This feature has been on my todo-list for quite some time...

Short answer: currently, no, there is no elegant way to pluralize acronyms.

The problem is that you will want to pluralize only some acronyms, not all of them; a style function was supposed to be used by all acronyms in a document. So we need to know for which occurrence we should pluralize, that is, to add some parameter.

The legacy markup `\acr{KEY}` uses a custom parsing, which is really not robust.
I was hoping that the new, Quarto-powered, shortcode markup `{{< acr KEY >}}` would allow me to add several new features that depend on parameters. They do simplify the parsing (for example, `{{< acr KEY pluralize >}}` is trivial to handle), but I cannot control when they are executed, which messes up the list of acronyms generation.

Until Quarto solves this on their end, I think I'll just have to update my parser to have something like `\acr{KEY}{pluralize}`.

Oct 16, 2024 10:05:52 Dylan Irion ***@***.***>:
